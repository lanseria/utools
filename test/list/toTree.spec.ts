import { listToTree } from "@lanseria/utools";
import { data4 } from "../mock";

describe("List To Tree 数组转为树操作", () => {
  it("normal 测试", () => {
    const tree = listToTree(data4 as IObj[]);
    console.log(tree);
    expect(tree).toMatchSnapshot();
  });
});
