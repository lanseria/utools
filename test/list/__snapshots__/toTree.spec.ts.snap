// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List To Tree 数组转为树操作 normal 测试 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "id": 2,
        "name": "a11",
        "parentId": 1,
      },
      Object {
        "children": Array [],
        "id": 3,
        "name": "a12",
        "parentId": 1,
      },
    ],
    "id": 1,
    "name": "a1",
    "parentId": 0,
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "id": 5,
        "name": "a21",
        "parentId": 4,
      },
      Object {
        "children": Array [],
        "id": 6,
        "name": "a22",
        "parentId": 4,
      },
    ],
    "id": 4,
    "name": "a2",
    "parentId": 0,
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "id": 8,
        "name": "a31",
        "parentId": 7,
      },
      Object {
        "children": Array [],
        "id": 9,
        "name": "a32",
        "parentId": 7,
      },
    ],
    "id": 7,
    "name": "a3",
    "parentId": 0,
  },
]
`;
